homeassistant/components/modbus/climate.py

```diff
@@ -412,9 +430,14 @@ async def async_set_swing_mode(self, swing_mode: str) -> None:
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        target_temperature = (
-            float(kwargs[ATTR_TEMPERATURE]) - self._offset
-        ) / self._scale
+        if self._target_temp_scale is not None and self._target_temp_offset is not None:
```

Comment:
this should be "or" not and....


but with this PR we are now trailing 3 times scale/offset through the component, that does not seem right, it would be a lot easier to eliminate the old scale/offset, by assigning those to target_ and current_ where you validate the parameters.

Reply:
Exactly as I wrote !! Do you seriously want me to write the code ?

Reply:
That would be the same as writing the code, it is your PR so it is your decision, there are no rules so you might choose another way that I would choose, which is perfectly OK.

I review what you suggest, and I have already told you what I look out for.